framework/src/packets/checksum.rs

```diff
@@ -1,8 +1,95 @@
 use common::Result;
-use packets::ip::IpAddrMismatchError;
-use std::net::IpAddr;
+use packets::ip::{IpAddrMismatchError, ProtocolNumber};
+use std::net::{IpAddr, Ipv4Addr, Ipv6Addr};
 use std::slice;
 
+/// Generic Pseudo Header used for checksumming purposes.
+pub struct PseudoHeader {
+    src: IpAddr,
+    dst: IpAddr,
+    packet_len: u16,
+    protocol: ProtocolNumber,
+}
+
+impl PseudoHeader {
+    pub fn new(src: IpAddr, dst: IpAddr, packet_len: u16, protocol: ProtocolNumber) -> Self {
```

Comment:
will move out of the default handler and impl the same fn for v4/v6 to assure it's right.

Reply:
discussed with @zeeshanlakhani maybe instead of validation, use an enum to represent only the valid states for a pseudo header where mismatched addrs won't be possible.